test/functional/terminal/ex_terminal_spec.lua

```diff
@@ -48,6 +49,31 @@ describe(':terminal', function()
     eq(2, eval("line('.')"))  -- cursor stays where we put it
   end)
 
+  it('does not crash when callback fires after terminal closes', function()
+    screen:detach()
+    screen = thelpers.screen_setup(0, '["'..helpers.nvim_prog..'", "-u", "NONE", "-i", "NONE", "-N"]')
+    thelpers.feed_data({':let g:split = 0',
+                        ':au TermOpen * nnoremap <buffer> x :q<CR>',
+                        ':nnoremap <silent> x :new <bar> let g:split += 1 <bar> call termopen([&shell])<CR>',
+                        ':set updatetime=0',
+                        ''})
```

Comment:
On some envs this is getting sent to the job stdin before the child nvim has a chance to receive it.

```
[ RUN      ] ...vim/neovim/test/functional/terminal/ex_terminal_spec.lua @ 52: :terminal does not crash when callback fires after terminal closes
./test/functional/ui/screen.lua:298: Failed to match any screen lines.
Expected (anywhere): "50                                                "
Actual:
  |:let g:split = 0                                  |
  |:au TermOpen * nnoremap <buffer> x :q<CR>         |
  |:nnoremap <silent> x :new <bar> let g:split += 1 <|
  |bar> call termopen([&shell])<CR>                  |
  |:set updatetime=0                                 |
  |{1: }                                                 |
  |{3:-- TERMINAL --}                                    |

stack traceback:
	./test/functional/ui/screen.lua:298: in function 'wait'
	./test/functional/ui/screen.lua:209: in function 'expect'
	...vim/neovim/test/functional/terminal/ex_terminal_spec.lua:74: in function <...vim/neovim/test/functional/terminal/ex_terminal_spec.lua:52>
```

In this case the most expedient way to avoid this would be to send the commands as `-c` args.